internal/component/discovery/discovery.go

```diff
@@ -76,34 +78,67 @@ func New(o component.Options, args component.Arguments, creator Creator) (*Compo
 
 // Run implements component.Component.
 func (c *Component) Run(ctx context.Context) error {
-	var cancel context.CancelFunc
+	var stopFn func() = nil
 	for {
 		select {
 		case <-ctx.Done():
 			return nil
```

Comment:
TODO: need to call stopFn here
I will do some more testing of this component as well to be sure